File: examples/typescript/policies/getPolicyById.ts
// Usage: pnpm tsx policies/getPolicyById.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();
const policy = await cdp.policies.createPolicy({
  policy: {
    scope: "account",
    description: "Project Policy",
    rules: [
      {
        action: "accept",
        operation: "signEvmTransaction",
        criteria: [
          {
            type: "ethValue",
            ethValue: "1000000000000000000",
            operator: "<=",
          },
          {
            type: "evmAddress",
            addresses: ["0x000000000000000000000000000000000000dEaD"],
            operator: "in",
          },
        ],
      },
    ],
  },
});

const retrievedPolicy = await cdp.policies.getPolicyById({
  id: policy.id,
});
console.log("Retrieved policy: ", JSON.stringify(retrievedPolicy, null, 2));

File: examples/python/policies/get_policy_by_id.py
# Usage: uv run python policies/get_policy_by_id.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv
from cdp.policies.types import CreatePolicyOptions, EthValueCriterion, SignEvmTransactionRule

load_dotenv()

async def main():
    async with CdpClient() as cdp:
        policy = CreatePolicyOptions(
            scope="account",
            description="Account Allowlist Example",
            rules=[
                SignEvmTransactionRule(
                    action="accept",
                    criteria=[
                        EthValueCriterion(
                            ethValue="1000000000000000000",
                            operator="<=",
                        ),
                    ],
                )
            ],
        )
        result = await cdp.policies.create_policy(policy=policy)
        print("Created policy: ", result)

        print("Retrieving policy by id: ", result.id)
        policy = await cdp.policies.get_policy_by_id(result.id)
        print("Retrieved policy: ", policy)


asyncio.run(main())
